### ISSUE TITLE
NCD clustering accuracy under traditional Settings 

### ISSUE THREAD

**Issue by ZhimaoPeng** [muliyangm/BYOP#3]
Thank you so much for sharing the code, it is very helpful for us !

I tried to run the code according to the command you provided in this repo, all the results look normal except for the traditional NCD clustering accuracy.

**Comment by muliyangm** [comment]
Hi Zhimao, could you pls share the details of your command and the checkpoint you used to reproduce the results? I have also uploaded the pretrained checkpoints I used for better reproducibility.

**Comment by ZhimaoPeng** [comment]
> Hi Zhimao, could you pls share the details of your command and the checkpoint you used to reproduce the results? I have also uploaded the pretrained checkpoints I used for better reproducibility.

For cifar 10-5 with imbalance ratio 100,  the command are as follows:

Pretraining stage: python main_pretrain.py --dataset CIFAR10 --data_dir /home/ubuntu/code/open-set/data --gpus 1 --precision 16 --max_epochs 200 --num_labeled_classes 5 --num_unlabeled_classes 5 --true_prior 100 --comment 5_5_p100 --offline

Discovery stage: python main_byop.py --dataset CIFAR100 --data_dir /home/ubuntu/code/open-set/data --gpus 1 --precision 16 --max_epochs 200 --num_labeled_classes 50 --num_unlabeled_classes 50 --true_prior 100 --pretrained checkpoints/pretrain-resnet18-CIFAR100-50_50_p100.cp --est_epoch 5 --comment run2

**Comment by muliyangm** [comment]
Though the commands are respectively for CIFAR10 and 100, they look good to me. Please note that the fluctuation in traditional clustering acc is not a rare case, which can be sensitive to code environments, computing devices and hyperparameters. Since you mentioned that results under other metrics are normal, you could try multiple runs and see how it goes.

**Comment by ZhimaoPeng** [comment]
> Though the commands are respectively for CIFAR10 and 100, they look good to me. Please note that the fluctuation in traditional clustering acc is not a rare case, which can be sensitive to code environments, computing devices and hyperparameters. Since you mentioned that results under other metrics are normal, you could try multiple runs and see how it goes.

The problem has to be solved.

**Comment by ZhimaoPeng** [comment]
> > Though the commands are respectively for CIFAR10 and 100, they look good to me. Please note that the fluctuation in traditional clustering acc is not a rare case, which can be sensitive to code environments, computing devices and hyperparameters. Since you mentioned that results under other metrics are normal, you could try multiple runs and see how it goes.
> 
> The problem has to be solved, thanks！

**Comment by ZhimaoPeng** [comment]
> > > Though the commands are respectively for CIFAR10 and 100, they look good to me. Please note that the fluctuation in traditional clustering acc is not a rare case, which can be sensitive to code environments, computing devices and hyperparameters. Since you mentioned that results under other metrics are normal, you could try multiple runs and see how it goes.
> > 
> > 
> > The problem has been solved, thanks！

**Comment by ZhimaoPeng** [comment]
> > > > Though the commands are respectively for CIFAR10 and 100, they look good to me. Please note that the fluctuation in traditional clustering acc is not a rare case, which can be sensitive to code environments, computing devices and hyperparameters. Since you mentioned that results under other metrics are normal, you could try multiple runs and see how it goes.
> > > 
> > > 
> > > The results do fluctuate sometimes, but normal results can be obtained, the problem has been solved, thanks！
